packages/next/package.json

```diff
@@ -53,7 +53,8 @@
     "prepublish": "npm run release && yarn types",
     "types": "tsc --declaration --emitDeclarationOnly --declarationDir dist",
     "typescript": "tsc --noEmit --declaration",
-    "ncc-compiled": "ncc cache clean && taskr ncc"
+    "ncc-compiled": "ncc cache clean && taskr ncc",
+    "build-native": "napi build --platform --release --cargo-cwd build/swc"
```

Comment:
Looking for feedback on how we should handle building the rust code. Currently it takes almost 5 minutes to compile on an M1 MacBook Pro and obviously requires rust to be installed. Should we add this command to the taskfile and require contributors to install Rust?

Reply:
Maybe we can pre-compile them and commit them to the repository similar to how we do for the `next/compiled` packages and add a check to CI to ensure they get re-compiled/committed when relevant files change.

Reply:
One issue is that we need to build for multiple platforms. I tried setting a different target than my local machine, but it failed to build with some confusing output. @kdy1 is it possible to expect contributors working on the rust code to build for all of the platforms and check it in? I found this issue which makes it seem like it's not supported, but I'm not sure what the `--target` cli option is for then (https://github.com/napi-rs/napi-rs/issues/273).

Reply:
Rust does not have an out-of-the-box cross compile mechanism, which `Go` and `Zig` provided. So it's really hard to build all platform/cpu targets on one system. The root cause of Rust's lack of a cross compile mechanism is that Rust relies on the `GCC` toolchain for the Link phase of compilation. And the GCC toolchain usually looks for the appropriate C dependencies on the host (glibc/musl-libc) for the system call on ***target*** platform/cpu. The toolchain and `C dependencies` must be set up properly on your host machine if you want cross compile Rust program.

[package-template](https://github.com/napi-rs/package-template) is the template project I set up for napi-rs users to build multi platform targets addons smoothly. And I've set up the cross build matrix and toolchain in it for the most used platforms: https://github.com/napi-rs/package-template/blob/main/.github/workflows/CI.yaml#L17-L198
The Docker images and used in this project is maintained by `napi-rs`, and tested by many projects.

If you have any questions about cross compile or `napi-rs`, feel free to contact with me! I'm willing to help here 🚀

Reply:
Thanks for the explanation and template! I just added a workflow based on that yesterday 🙂  https://github.com/vercel/next.js/pull/27355/commits/6aa57483d321c28eac3290600dcd6ed493629db2